cmd/fossa/cmd/test/test.go

```diff
@@ -112,44 +104,39 @@ func CheckBuild(locator fossa.Locator, stop <-chan time.Time) (fossa.Build, erro
 		case <-stop:
 			return fossa.Build{}, errors.New("timed out while waiting for build")
 		default:
-			builds, err := fossa.GetBuilds(locator)
+			build, err := fossa.GetLatestBuild(locator)
 			if _, ok := err.(api.TimeoutError); ok {
 				time.Sleep(pollRequestDelay)
 				continue
 			}
 			if err != nil {
 				return fossa.Build{}, errors.Wrap(err, "error while loading build")
 			}
-			if len(builds) == 0 {
-				time.Sleep(pollRequestDelay)
-				continue
-			}
-			latestBuild := builds[0]
-			switch latestBuild.Task.Status {
+			switch build.Task.Status {
 			case "SUCCEEDED":
-				return latestBuild, nil
+				return build, nil
 			case "FAILED":
-				return latestBuild, fmt.Errorf("failed to analyze build #%d: %s (visit FOSSA or contact support@fossa.io)", latestBuild.ID, latestBuild.Error)
+				return build, fmt.Errorf("failed to analyze build #%d: %s (visit FOSSA or contact support@fossa.io)", build.ID, build.Error)
 			default:
```

Comment:
I would like to change this so that we continue polling on a specific build status (e.g. "IN PROGRESS"?) instead of polling by default. This way we can _return an error_ by default (i.e. on unrecognised statuses) so we can detect bugs earlier.

Reply:
The set of possible statuses is `"CREATED" | "ASSIGNED" | "RUNNING" | "SUCCEEDED" | "FAILED"`, and we'd want to keep polling on anything other than succeeded or failed.

Reply:
Yes, but one day that might change. When it does, I want an error to occur instead of infinite polling.